Percolation.java

```diff
@@ -1,14 +1,161 @@
+import edu.princeton.cs.algs4.StdOut;
 import edu.princeton.cs.algs4.StdRandom;
 import edu.princeton.cs.algs4.StdStats;
 import edu.princeton.cs.algs4.WeightedQuickUnionUF;
 
 public class Percolation {
-   public Percolation(int n)                // create n-by-n grid, with all sites blocked
-   public    void open(int row, int col)    // open site (row, col) if it is not open already
-   public boolean isOpen(int row, int col)  // is site (row, col) open?
-   public boolean isFull(int row, int col)  // is site (row, col) full?
-   public     int numberOfOpenSites()       // number of open sites
-   public boolean percolates()              // does the system percolate?
+    
+    // gridLength is the length of the square grid. 
+    // There are gridLength^2 non-virtual sites in the grid
+    private int gridLength;
+    
+     // two sites can percolate to one another if they are both open and connected:
+    // isOpen monitors open/closed state of each site.
+    private boolean[] isOpen;
+    
+    // percolation represents connectivity between sites. connected, open sites percolate to one another.
+    private WeightedQuickUnionUF percolation;
+    
+    // quick union structure for tracking fullness without backwash.
+    // similar to percolation above, but without bottom virtual site
+    private WeightedQuickUnionUF fullness;
+    
+    // index of virtual site that is connected to entire top row, initializes to 0.
+    private int virtualTopIndex; 
+    
+    // index of virtual site that is connected to entire bottom row, initializes to (n^2)+1
+    private int virtualBottomIndex;
+    
+    // converts between two dimensional coordinate system and site array index. 
+    // throws exceptions on invalid bounds. valid indices are 1 : n^2
+    // a is the row; b is the column
+    private int siteIndex(int a, int b)
```

Comment:
naming your variables row and column instead of a and b will make your code much more readable. Please could you update this code to use row and column instead of a and b?